### ISSUE TITLE
Loss of synchronization in blocks for working with a List Viewer (Live, Device)

### ISSUE THREAD

**Issue by AlexITMan** [thunkable/thunkable-issues#552]
When executing blocks in the drawing, Column1.Height = 0 and Label1. Text = 0

![изображение](https://user-images.githubusercontent.com/20015046/83727095-3760ba00-a64d-11ea-90dc-cd38ae00ad5d.png)

**Comment by kezb765** [comment]
Hi @AlexITMan , maybe I had a similar problem, have a look at issue #534. I suspect that actions like "set text" or "set textitems" are now executed asynchronous. This used to be a synchronous action IMHO.

**Comment by AlexITMan** [comment]
Hi, yes, these are similar problems. Green blocks must be synchronous. If the block is asynchronous, it must have a block with a callback function or a "then do" area.

**Comment by jaredgib** [comment]
im experiencing the same issues. synchronous when i use the live preview but not when i publish to the web. super odd

**Comment by AlexITMan** [comment]
There is nothing strange about this. This is because the application project runs in different environments. Browser browsing is one runtime environment, and running an app in an Android emulator is another runtime environment, and Live on Android is the third environment, and apk on Android is the fourth environment. These environments are very similar, but they are not identical and have their own characteristics. If a project works perfectly in one of these environments, it does not mean that it will work in all the others as well. I think you can give examples of this yourself.

**Comment by jaredgib** [comment]
Yes, good point. Thank you, This is true. Sometimes, because I am still new to programming, I forget those variables also affect how well the app code will be executed.

**Comment by jane-d-alt** [comment]
Hi there,

It looks like this behavior has mostly been resolved. When I use similar blocks in [this project](https://x.thunkable.com/projectPage/60a4c92c5062c605e6af5bf8/), the text items and length of the list can be printed immediately. 

The Computed Height does not update immediately, and still requires a wait block. I will let the team know about this behavior. 

Thanks for reporting, and apologies for not following up here when the synchronicity of the other blocks was restored.
